hypothesis-python/src/hypothesis/_settings.py

```diff
@@ -209,34 +207,50 @@ def all(cls) -> list["HealthCheck"]:
     which is enabled by default when running Hypothesis inside pytest."""
 
     differing_executors = 10
-    """Checks if :func:`@given <hypothesis.given>` has been applied to a test
+    """Checks if |@given| has been applied to a test
     which is executed by different :ref:`executors<custom-function-execution>`.
     If your test function is defined as a method on a class, that class will be
     your executor, and subclasses executing an inherited test is a common way
     for things to go wrong.
 
     The correct fix is often to bring the executor instance under the control
     of hypothesis by explicit parametrization over, or sampling from,
-    subclasses, or to refactor so that :func:`@given <hypothesis.given>` is
+    subclasses, or to refactor so that |@given| is
     specified on leaf subclasses."""
 
     nested_given = 11
-    """Checks if :func:`@given <hypothesis.given>` is used inside another
-    :func:`@given <hypothesis.given>`. This results in quadratic generation and
+    """Checks if |@given| is used inside another
+    |@given|. This results in quadratic generation and
     shrinking behavior, and can usually be expressed more cleanly by using
     :func:`~hypothesis.strategies.data` to replace the inner
-    :func:`@given <hypothesis.given>`.
+    |@given|.
 
     Nesting @given can be appropriate if you set appropriate limits for the
     quadratic behavior and cannot easily reexpress the inner function with
     :func:`~hypothesis.strategies.data`. To suppress this health check, set
     ``suppress_health_check=[HealthCheck.nested_given]`` on the outer
-    :func:`@given <hypothesis.given>`. Setting it on the inner
-    :func:`@given <hypothesis.given>` has no effect. If you have more than one
+    |@given|. Setting it on the inner
+    |@given| has no effect. If you have more than one
     level of nesting, add a suppression for this health check to every
-    :func:`@given <hypothesis.given>` except the innermost one.
+    |@given| except the innermost one.
     """
 
+    @classmethod
+    def all(cls) -> list["HealthCheck"]:
+        """
+        Returns a list of all |HealthCheck| enum values. Can be used with
+        ``suppress_health_check=HealthCheck.all()`` to suppress all health
+        checks - though we recommend silencing individual health checks instead!
+        """
+        # Skipping of deprecated attributes is handled in HealthCheckMeta.__iter__
+        note_deprecation(
+            "`HealthCheck.all()` is deprecated; use `list(HealthCheck)` instead.",
+            since="2023-04-16",
+            has_codemod=True,
+            stacklevel=1,
+        )
```

Comment:
I'd prefer to omit deprecated methods from the documentation, or at least explicitly note that it's deprecated + the recommended alternative.

Reply:
oh I actually forgot this was deprecated. I'll just remove it.